### DISCUSSION TITLE
Classnames with `next/link` should be forwarded to the `a` tag

### DISCUSSION THREAD

**Discussion by discoverlance-com** [vercel/next.js#38789] [Ideas]
In recent Next.JS update to the `next/link` component, it auto renders an `a` tag which means we can use `next/link` without a nested `a` tag as specified in the example below
```js
import Link from next/link

<Link href="/">Homepage</Link>
// this outputs <a href="/">Homepage</a>
```

When I add *className* prop to the *Link* tag, it's not passed to the generated `a` tag. I think this should be the default as the solution I tried at the moment is to use the old way of adding a  nested `a` tag in the Link component and adding the className to it. So it seems even though the className prop is accepted on the *Link* tag, it's useless as it's not even passed to the *a* tag when nested.

```js
// this does not pass the className prop to the generated link tag
<Link href="/" className="flex justify-center">Homepage</Link>
// this still outputs <a href="/">Homepage</a>

// this also does not pass the className prop to the generated link tag
<Link href="/" className="flex justify-center"> 
  <a>Homepage</a>
</Link>
// this still outputs <a href="/">Homepage</a>

// How to go around it
<Link href="/" > 
  <a className="flex justify-center">Homepage</a>  
</Link>
// this outputs <a href="/" className="flex justify-center">Homepage</a>


// Proposed way
<Link href="/" className="flex justify-center">Homepage</Link>
// I think it should output this <a href="/" className="flex justify-center">Homepage</a>
```

**Comment by valcosmos** [comment]
Hello, Here is an example about active classname on `Link`, hope it can help you.

[Active className on Link](https://github.com/vercel/next.js/tree/canary/examples/active-class-name)

**Reply by discoverlance-com** [reply]
> Hello, Here is an example about active classname on `Link`, hope it can help you.
> 
> [Active className on Link](https://github.com/vercel/next.js/tree/canary/examples/active-class-name)

The active className I think helps when you want to set the active classes on the link which  does not really apply to the use case I mentioned where the className prop on `next/link` is not useful and I think it could instead be passed unto the `<a>` tag that is generated when using `next/link`
